lib/mergeusertool.php

```diff
@@ -296,6 +314,14 @@ private function _merge($toid, $fromid)
                     $e->getTraceAsString() . html_writer::empty_tag('br'));
         }
 
+        if ($this->alwaysRollback) {
+            $transaction->rollback(new Exception('alwaysRollback option is set so rolling back transaction'));
+        }
+
+        if ($this->debugdb) {
```

Comment:
To make things correct, and respect the workflow, I think this condition should be placed before the above `if ($this->alwaysRollback) {`, so that, if the transaction should finish, we are sure there will not be any further message about the DB if any.